### PR TITLE
chore: deduplicate library verificaation code

### PR DESCRIPTION
**Pull request by Vaelatern**

### CODE REVIEW

**Comment by bh90210** [comment]
hey @Vaelatern, what exactly happened here?

**Comment by Vaelatern** [comment]
Someone pointed out low hanging fruit for deduplication, so I grabbed it.

File: cmd/linux.go [hunk 1]
```diff
@@ -166,9 +169,9 @@ func RpmInstalled(packageName string) (bool, error) {
 
 // RequestSupportForDistribution promts the user to submit a request to support their
 // currently unsupported distribution
-func RequestSupportForDistribution(distroInfo *DistroInfo, libraryName string) error {
+func RequestSupportForDistribution(distroInfo *DistroInfo) error {
 	var logger = NewLogger()
-	defaultError := fmt.Errorf("unable to check libraries on distribution '%s'. Please ensure that the '%s' equivalent is installed", distroInfo.Name, libraryName)
+	defaultError := fmt.Errorf("unable to check libraries on distribution '%s'.", distroInfo.Name)
```
Comment (bh90210) [inline cmd/linux.go]: this change makes sense. Seems redundant here to give back the message "Please ensure that the '%s' equivalent is installed"

File: cmd/linux.go [hunk 2]
```diff
@@ -120,6 +120,9 @@ func parseOsRelease(osRelease string) *DistroInfo {
 	return result
 }
 
+// GetPkgInstalled is all functions that use local programs to see if a package is installed
+type GetPkgInstalled func(string) (bool, error)
```
Comment (bh90210) [inline cmd/linux.go]: may I suggest changing this to `CheckPkgInstalled`? It is a bit confusing as it seems to me that it'll get a pkg installed.

File: cmd/system.go [hunk 3]
```diff
@@ -269,55 +269,33 @@ func CheckDependencies(logger *Logger) (bool, error) {
 		if err != nil {
 			return false, err
 		}
+
+		var libraryChecker GetPkgInstalled
```
Comment (bh90210) [inline cmd/system.go]: code became clearer indeed. :+1:

**Comment by bh90210** [comment]
```bash
./0_setup.go:111:52: too many arguments in call to cmd.RequestSupportForDistribution
        have (*cmd.DistroInfo, string)
        want (*cmd.DistroInfo)
```
please go to [wails/cmd/wails/0_setup.go](https://github.com/Vaelatern/wails/blob/deduplicate-code/cmd/wails/0_setup.go) line 111 and replace the line with `return false, cmd.RequestSupportForDistribution(distroInfo)`

**Comment by leaanthony** [comment]
This is fantastic @Vaelatern! Thank you. The Linux support has probably evolved quicker than any other part so it is ripe for tidying up. 👍

**Comment by leaanthony** [comment]
Do you want to include Void into this PR?

File: cmd/linux.go (hunk 1 above)
Comment (leaanthony) [inline cmd/linux.go]: Any chance you could remove the full stop at the end of the message? Codacy is moaning :-)

Comment (Vaelatern) [reply cmd/linux.go]: Done!

**Comment by Vaelatern** [comment]
Fixed the error, @bh90210 

@leaanthony I'd rather rebase the Void support on top of this after it's in, if that's OK.

**Comment by leaanthony** [comment]
@Vaelatern awesome first PR! 👍

**Pull request merged**